src/librustc_parse/parser/item.rs

```diff
@@ -111,13 +124,33 @@ impl<'a> Parser<'a> {
             .emit();
     }
 
+    /// Error in-case a `default` was parsed but no item followed.
+    fn error_on_unmatched_defaultness(&self, def: Defaultness) {
+        if let Defaultness::Default(span) = def {
+            self.struct_span_err(span, "unmatched `default`")
+                .span_label(span, "the unmatched `default`")
+                .emit();
+        }
+    }
+
+    /// Error in-case `default` was parsed in an in-appropriate context.
+    fn error_on_illegal_default(&self, def: Defaultness) {
+        if let Defaultness::Default(span) = def {
+            self.struct_span_err(span, "item cannot be `default`")
```

Comment:
Items can be default (syntactically) if they are `fn`, `const` or `type` items.

Reply:
Highly related to https://github.com/rust-lang/rust/pull/69366#discussion_r382901198.

Reply:
You're saying you'd like to syntactically allow `extern { default fn foo(); }` as well, right?

Reply:
Adjusted based on the assumption that ^-- is accurate.